### PR TITLE
Add log when tappers cannot be started

### PR DESCRIPTION
**Pull request by gadotroee**

### CODE REVIEW

File: shared/kubernetes/mizuTapperSyncer.go [hunk 1]
```diff
@@ -310,6 +311,22 @@ func (tapperSyncer *MizuTapperSyncer) updateMizuTappers() error {
...
 	logger.Log.Debugf("Updating DaemonSet to run on nodes: %v", nodesToTap)
 
+	if nodes, err := tapperSyncer.kubernetesProvider.GetNodes(tapperSyncer.context, ""); err == nil {
+		for _, item := range nodes.Items {
+			if sliceutil.Contains(nodesToTap, item.Name) {
+				val, ok := item.GetLabels()[NodeHostNameLabelKey]
+				if !ok {
+					logger.Log.Warningf("Tapper cannot start on node '%s', this missing the '%s' label", item.Name, NodeHostNameLabelKey)
+				} else if val != item.Name {
+					logger.Log.Warningf("Tapper cannot start on node '%s',  value of '%s' label is not the node name", item.Name, NodeHostNameLabelKey, val)
```
Comment (IgorGov) [inline shared/kubernetes/mizuTapperSyncer.go:321]:
```suggestion
					logger.Log.Warningf("Tapper cannot start on node '%s',  value of '%s' label is not match the node name", item.Name, NodeHostNameLabelKey, val)
```

File: shared/kubernetes/mizuTapperSyncer.go [hunk 2]
```diff
@@ -310,6 +311,22 @@ func (tapperSyncer *MizuTapperSyncer) updateMizuTappers() error {
 
 	logger.Log.Debugf("Updating DaemonSet to run on nodes: %v", nodesToTap)
 
+	if nodes, err := tapperSyncer.kubernetesProvider.GetNodes(tapperSyncer.context, ""); err == nil {
+		for _, item := range nodes.Items {
+			if sliceutil.Contains(nodesToTap, item.Name) {
```
Comment (IgorGov) [inline shared/kubernetes/mizuTapperSyncer.go:316]: is there a possibility that one of the 'nodesToTap' won't be in the 'nodes.Items', do we want to handle it?

Comment (gadotroee) [reply shared/kubernetes/mizuTapperSyncer.go]: handled it better in https://github.com/up9inc/mizu/pull/1018/commits/92e2ea89dd5d807470d93b5123636b069ea9b1e1

File: shared/kubernetes/mizuTapperSyncer.go [hunk 3]
```diff
@@ -358,4 +360,4 @@ func (tapperSyncer *MizuTapperSyncer) updateMizuTappers() error {
 	tapperSyncer.tappedNodes = nodesToTap
 
 	return nil
-}
+}
```
Comment (IgorGov) [inline shared/kubernetes/mizuTapperSyncer.go:363]: new line

File: shared/kubernetes/utils.go [hunk 4]
```diff
@@ -93,3 +95,29 @@ func GetPodInfosForPods(pods []core.Pod) []*shared.PodInfo {
...
+			nodeNameToDetails[item.Name] = item
+		}
+	} else {
+		logger.Log.Debugf("cannot check nodes of cluster")
+	}
+
+	for _, item := range nodesToTap {
+		nodeDetails, ok := nodeNameToDetails[item]
+		if !ok {
+			logger.Log.Debugf("Cannot get this node details to check if the tapper can start on it")
```
Comment (IgorGov) [inline shared/kubernetes/utils.go:113]: Lets add more details to the logs
```suggestion
			logger.Log.Debugf("Failed to find node to tap %v in cluster nodes %v", item, list(nodeNamesToDetails.keys))
```

File: shared/kubernetes/utils.go [hunk 5]
```diff
@@ -93,3 +95,29 @@ func GetPodInfosForPods(pods []core.Pod) []*shared.PodInfo {
...
+			continue
+		}
+		val, ok := nodeDetails.GetLabels()[NodeHostNameLabelKey]
+		if !ok {
+			logger.Log.Warningf("Tapper cannot start on node '%s', this missing the '%s' label", nodeDetails.Name, NodeHostNameLabelKey)
```
Comment (IgorGov) [inline shared/kubernetes/utils.go:118]:
```suggestion
			logger.Log.Warningf("Failed to tap node %s: node label %s wasn't found", item, NodeHostNameLabelKey)
```

File: shared/kubernetes/utils.go [hunk 6]
```diff
@@ -93,3 +95,29 @@ func GetPodInfosForPods(pods []core.Pod) []*shared.PodInfo {
...
+		} else if val != nodeDetails.Name {
+			logger.Log.Warningf("Tapper cannot start on node '%s',  value of '%s' label is not match the node name", nodeDetails.Name, NodeHostNameLabelKey, val)
```
Comment (IgorGov) [inline shared/kubernetes/utils.go:120]:
```suggestion
			logger.Log.Warningf("Failed to tap node %s:,  node label '%s' is not matching the node name", nodeDetails.Name, NodeHostNameLabelKey, val)
```

File: shared/kubernetes/utils.go [hunk 7]
```diff
@@ -93,3 +95,29 @@ func GetPodInfosForPods(pods []core.Pod) []*shared.PodInfo {
 	}
 	return podInfos
 }
+
+
+func checkTappersCanStartOnNodes(kubernetesProvider *Provider, context context.Context, nodesToTap []string) {
+	var nodeNameToDetails = make(map[string]core.Node, 0)
```
Comment (gadotroee) [inline shared/kubernetes/utils.go:101]:
```suggestion
	var nodeNameToDetails = make(map[string]core.Node, 0)
	var clusterNodeNames = make([]string, 0)
```

File: shared/kubernetes/utils.go [hunk 8]
```diff
@@ -93,3 +95,29 @@ func GetPodInfosForPods(pods []core.Pod) []*shared.PodInfo {
...
+	if nodes, err := kubernetesProvider.GetNodes(context, ""); err == nil {
+		for _, item := range nodes.Items {
+			nodeNameToDetails[item.Name] = item
```
Comment (gadotroee) [inline shared/kubernetes/utils.go:104]:
```suggestion
			nodeNameToDetails[item.Name] = item
			clusterNodeNames = append(clusterNodeNames, item.Name)
```

**Comment by gadotroee** [comment]
I don't think we need this after https://github.com/up9inc/mizu/pull/1022

**Pull request closed without merging**